lib/db_connection.ex

```diff
@@ -1071,9 +1087,11 @@ defmodule DBConnection do
     end
   end
 
-  defp encode(query, params, meter, opts) do
+  defp maybe_encode(query, params, meter, opts) do
     try do
       DBConnection.Query.encode(query, params, opts)
+    rescue
+      DBConnection.EncodeError -> :retry
```

Comment:
For clarity could we rename `:retry` to be `:prepare`, as it is not a straight retry, we have to do some extra work (prepare). Also we should return the existing `meter`, so its `{:prepare, meter}` so that we create it once.